src/modules/roc_sndio/target_sox/roc_sndio/sox_backend.cpp

```diff
@@ -175,6 +166,51 @@ void log_handler(unsigned sox_level,
     roc_log(level, "sox: %s: %s", filename, message);
 }
 
+void add_driver_mapping(core::Array<DriverInfo>& list,
+                        const char* driver,
+                        IBackend* backend,
+                        unsigned int driver_flags) {
+    for (size_t n = 0; n < list.size(); n++) {
+        if (strcmp(list[n].name, driver) == 0 && list[n].backend == backend) {
+            return;
+        }
+    }
+    DriverInfo driver_info;
+    driver_info.set(driver, backend, driver_flags);
+    list.push_back(driver_info);
+}
+
+template <class T>
+bool check_and_open(const char* driver,
+                    const char* inout,
+                    int filter_flags,
+                    const core::ScopedPtr<T>& sinksource) {
+    if (driver && is_driver_hidden(driver)) {
```

Comment:
It seems it can't happen with the new backend dispatcher algorithm: if driver is non-null, backend dispatcher will use only backend that explicitly specified it supports given driver. So we can remove this check.